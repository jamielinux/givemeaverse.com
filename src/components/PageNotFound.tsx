/* SPDX-License-Identifier: MPL-2.0 */

/* Copyright © 2020 Jamie Nguyen
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/. */

import React from 'react';
import { Link } from 'react-router-dom';
import { Helmet } from 'react-helmet';

import Header from 'components/common/Header';

const PageNotFound = (): React.ReactElement => {
  return (
    <>
      <Helmet>
        <title>404 — Page Not Found</title>
        <meta name="description" content="Page not found." />
      </Helmet>

      <Header format="static" />

      <main id="main" className="page-not-found">
        <section className="page-not-found__content">
          <h1 className="page-not-found__title">404 — Page Not Found</h1>
          <p className="page-not-found__text">
            Sorry, this page doesn’t exist! Go back to the{' '}
            <Link to="/" className="clickable  page-not-found__link">
              home page
            </Link>
            .
          </p>
        </section>
      </main>
    </>
  );
};

export default PageNotFound;
