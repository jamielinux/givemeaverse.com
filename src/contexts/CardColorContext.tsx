/* SPDX-License-Identifier: MPL-2.0 */

/* Copyright Â© 2020 Jamie Nguyen
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/. */

import React from 'react';
import PropTypes from 'prop-types';
import { useLocation } from 'react-router-dom';
import { NUM_VERSE_COLORS } from 'constants/index';
import { createCtx } from 'utils/index';

type CardColorContextType = {
  verseCardColor: number;
};

export type LocationStateType = {
  verseCardColor?: number;
  avoidColor?: number;
  avoidVerseId?: string;
};

const [useCardColorContext, ContextProvider] = createCtx<
  CardColorContextType
>();

export function CardColorContextProvider({
  children,
}: {
  children: React.ReactNode;
}): React.ReactElement {
  const { state }: { state: LocationStateType } = useLocation();

  // Avoid using the same color twice in a row.
  function getRandomColor(): number {
    if (state && typeof state.verseCardColor !== 'undefined') {
      // If just switching translation, keep the same verseCardColor.
      return state.verseCardColor;
    }

    if (
      typeof state !== 'undefined' &&
      typeof state.avoidColor !== 'undefined'
    ) {
      // If there's a previous color, pick a random color except that one.
      const filteredColors = Array.from(Array(NUM_VERSE_COLORS).keys()).filter(
        (color) => color !== state.avoidColor,
      );
      return filteredColors[Math.floor(Math.random() * filteredColors.length)];
    }

    // If no previous color, return any random color.
    return Math.floor(Math.random() * NUM_VERSE_COLORS);
  }
  const verseCardColor = getRandomColor();
  const value = { verseCardColor };
  return (
    <ContextProvider value={value as CardColorContextType}>
      {children}
    </ContextProvider>
  );
}

CardColorContextProvider.propTypes = {
  children: PropTypes.node.isRequired,
};

export { useCardColorContext };
