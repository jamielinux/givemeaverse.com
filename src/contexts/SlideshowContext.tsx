/* SPDX-License-Identifier: MPL-2.0 */

/* Copyright Â© 2020 Jamie Nguyen
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/. */

import React, { useRef, useState, useLayoutEffect, useEffect } from 'react';
import { useHistory, useParams } from 'react-router-dom';
import PropTypes from 'prop-types';
import { SUPPORTED_SLIDESHOW_INTERVALS } from 'constants/index';
import { useCardColorContext } from 'contexts/CardColorContext';
import {
  createCtx,
  unhideModal,
  readValidateStorage,
  pushRandomVerse,
} from 'utils/index';

type SlideshowContextType = {
  slideshowInterval: number;
  applySlideshowInterval: (requestedInterval: number) => void;
  pauseSlideshow: () => void;
  restartSlideshow: (interval?: number) => void;
};

const [useSlideshowContext, ContextProvider] = createCtx<
  SlideshowContextType
>();

export function SlideshowContextProvider({
  children,
}: {
  children: React.ReactNode;
}): React.ReactElement {
  const [slideshowInterval, setSlideshowInterval] = useState(() => {
    const storedSetting = readValidateStorage({
      name: 'slideshowInterval',
      supportedSettings: SUPPORTED_SLIDESHOW_INTERVALS,
      defaultValue: '-1',
    });
    return Number(storedSetting);
  });

  const history = useHistory();
  const { verseId } = useParams();
  const { verseCardColor } = useCardColorContext();
  const countdownRef = useRef(slideshowInterval);
  const prevSlideshowInterval = useRef(slideshowInterval);
  const prevCountdownRef = useRef(-1);

  function applySlideshowInterval(requestedInterval: number): void {
    setSlideshowInterval(requestedInterval);
  }

  function pauseSlideshow(): void {
    prevSlideshowInterval.current = slideshowInterval;
    prevCountdownRef.current = countdownRef.current;
    countdownRef.current = -1;
  }

  function restartSlideshow(interval?: number): void {
    if (
      typeof interval !== 'undefined' &&
      interval !== prevSlideshowInterval.current
    ) {
      // User may have changed the slideshow interval in the slideshow dialog.
      countdownRef.current = interval;
    } else {
      countdownRef.current = prevCountdownRef.current;
    }
  }

  useLayoutEffect(() => {
    // ACCESSIBILITY: Hide elements from tab select and screen readers.
    // If the slideshow popup is open, re-hide elements after re-render.
    const content = document.getElementById('content');
    const modal = document.getElementById('slideshow');
    if (modal && !modal.hasAttribute('aria-hidden')) {
      unhideModal({ content, modal });
    }
  }, [slideshowInterval]);

  // Automatically select another verse if slideshow is enabled.
  useEffect(() => {
    const interval = setInterval(() => {
      if (countdownRef.current >= 0) {
        countdownRef.current -= 1;
      }
      if (countdownRef.current === 0) {
        countdownRef.current = slideshowInterval;
        const state = {
          avoidColor: verseCardColor,
          avoidVerseId: verseId,
        };
        pushRandomVerse({ history, state });
      }
    }, 1000);
    return (): void => clearInterval(interval);
  }, [history, slideshowInterval, verseCardColor, verseId]);

  useEffect(() => {
    localStorage.setItem('slideshowInterval', slideshowInterval.toString());
  }, [slideshowInterval]);

  const value = {
    slideshowInterval,
    applySlideshowInterval,
    pauseSlideshow,
    restartSlideshow,
  };

  return (
    <ContextProvider value={value as SlideshowContextType}>
      {children}
    </ContextProvider>
  );
}

SlideshowContextProvider.propTypes = {
  children: PropTypes.node.isRequired,
};

export { useSlideshowContext };
