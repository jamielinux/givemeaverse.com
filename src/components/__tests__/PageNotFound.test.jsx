/* SPDX-License-Identifier: MPL-2.0 */

/* Copyright © 2020 Jamie Nguyen
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/. */

/* eslint-disable no-undef */

import React from 'react';
import { Helmet } from 'react-helmet';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import { render, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';

import { CardColorContextProvider } from 'contexts/CardColorContext';
import { SlideshowContextProvider } from 'contexts/SlideshowContext';
import PageNotFound from '../PageNotFound';

function component(history) {
  return (
    <Router history={history}>
      <CardColorContextProvider>
        <SlideshowContextProvider>
          <PageNotFound />
        </SlideshowContextProvider>
      </CardColorContextProvider>
    </Router>
  );
}

describe('PageNotFound page', () => {
  let history;
  let getByText;
  beforeEach(() => {
    history = createMemoryHistory();
    getByText = render(component(history)).getByText;
  });

  test('document title is correct', () => {
    const helmet = Helmet.peek();
    expect(helmet.title).toEqual('404 — Page Not Found');
  });

  test('SNAPSHOT: 404 page', () => {
    const tree = renderer.create(component(history));
    const treeJSON = tree.toJSON();
    expect(treeJSON).toMatchSnapshot();
  });

  test('link to home page can be clicked', () => {
    const link = getByText('home page');
    fireEvent.click(link);
    expect(history.location.pathname).toBe('/');
  });

  // eslint-disable-next-line jest/expect-expect
  test('all clickable elements have "clickable" class, except the nav', () => {
    const events = getAllEventListeners();
    expectClickable(events);
  });

  // eslint-disable-next-line jest/expect-expect
  test('menu is hidden to begin with', () => {
    expectNavIsInactive();
  });

  // eslint-disable-next-line jest/expect-expect
  test('menu can be opened and closed', () => {
    expectMenuCanBeOpenedAndClosed();
  });
});
