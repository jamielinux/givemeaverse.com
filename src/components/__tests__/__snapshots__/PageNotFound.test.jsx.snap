// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageNotFound page SNAPSHOT: 404 page 1`] = `
Array [
  <header>
    <div
      className="header"
    >
      <div
        className="header__logo"
      >
        <a
          aria-label="Logo. Click here to navigate to the home page."
          className="clickable  header__logo--static"
          href="/"
          onClick={[Function]}
          role="button"
          tabIndex={0}
        >
          <svg
            aria-hidden="true"
            className="svg-logo"
            height="30"
            viewBox="0 0 36.173 7.938"
            xmlns="http://www.w3.org/2000/svg"
          >
            <defs>
              <clipPath
                clipPathUnits="userSpaceOnUse"
                id="clip-path"
              >
                <path
                  d="M6.185 2.056a.33.33 0 01-.069-.139c-.02-.065-.02-.098 0-.098.031-.004.066-.037.105-.099.042-.065.061-.119.057-.161 0-.015.006-.029.017-.04a.076.076 0 01.041-.03.166.166 0 01.081-.028c.065 0 .125.046.18.139.023.05.034.093.034.132 0 .035-.01.071-.029.11a.903.903 0 01-.105.139c-.065.088-.127.132-.184.132-.039 0-.081-.019-.127-.057zm-.56 2.346c-.053-.054-.08-.131-.08-.232 0-.053.01-.1.028-.138 0-.02.004-.043.012-.07a.244.244 0 00.012-.052.576.576 0 01.046-.19c.069-.155.161-.426.277-.815 0-.02.008-.035.024-.047a.115.115 0 01.068-.028.346.346 0 01.064-.011c.023 0 .043.01.058.028.02.016.044.047.075.092.023.023.034.058.034.105a.632.632 0 01-.075.265c-.05.112-.083.208-.098.289-.047.104-.083.2-.11.289-.007.023-.018.07-.034.139a.553.553 0 01-.064.167 1.23 1.23 0 01-.063.226c-.02.042-.041.063-.064.063-.027 0-.063-.027-.11-.08zm1.951 0c-.039.027-.064.04-.075.04-.015 0-.046-.023-.093-.07-.069-.07-.119-.198-.15-.387a5.655 5.655 0 01-.046-.832c.02-.208.029-.406.029-.595.012-.015.035-.023.07-.023.038 0 .07.008.097.023.039.02.064.048.076.087a.794.794 0 01.029.225c.003.112.005.255.005.428a5.593 5.593 0 00-.017.404.57.57 0 00.017.191c.016.027.034.035.053.024a1.07 1.07 0 00.12-.11l.227-.341c.019-.02.053-.066.103-.139l.092-.115a.938.938 0 01.058-.082.365.365 0 00.058-.086c.043-.07.087-.13.132-.18.047-.05.078-.074.094-.074a.11.11 0 00.028.011.09.09 0 00.029-.006.106.106 0 01.029-.005c.015-.016.031-.024.047-.024.023 0 .044.008.063.024.038.03.057.057.057.08 0 .031-.03.103-.092.214-.061.108-.105.17-.132.185l-.059.058c-.011.05-.082.163-.213.34-.127.177-.255.347-.382.51a.918.918 0 01-.254.224zm2.223.202a.518.518 0 01-.41-.185.675.675 0 01-.163-.463c0-.185.04-.387.122-.606.085-.224.193-.424.324-.6.13-.177.263-.297.397-.36a.547.547 0 01.51 0c.065.027.115.06.15.1a.498.498 0 01.092.179c.011.069.017.116.017.143 0 .023-.01.073-.029.15a1.21 1.21 0 01-.184.365.971.971 0 01-.289.248.43.43 0 01-.232.058.668.668 0 01-.23-.04.384.384 0 01-.163-.133l-.04-.046-.017.046c-.08.208-.121.385-.121.53 0 .1.019.182.057.244.027.042.093.063.197.063a.451.451 0 00.168-.023.677.677 0 00.15-.069c.047-.031.09-.062.132-.093l.023-.011h-.005c.047-.023.085-.045.115-.064l.064-.04a.058.058 0 00.024-.006.08.08 0 00.022-.023c.035-.015.075-.01.121.017.047.027.07.062.07.105a.48.48 0 01-.104.15c-.062.07-.108.106-.139.11a.206.206 0 00-.04.017.189.189 0 00-.046.028h.005c-.035.047-.07.07-.104.07h-.047c-.015.03-.067.062-.156.093a.702.702 0 01-.241.046zm.26-1.272c.115 0 .221-.065.317-.196a.614.614 0 00.139-.37c0-.065-.013-.112-.04-.139a.148.148 0 00-.064-.017.478.478 0 00-.098.017c-.05.012-.127.066-.231.163a1.455 1.455 0 00-.232.248l-.075.121a.257.257 0 01.087-.017c.031-.004.049-.002.052.006a.21.21 0 00.04.132.125.125 0 00.104.052zm5.89.751c.015.012.022.03.022.053 0 .027-.01.053-.03.08-.015.023-.03.035-.046.035a.434.434 0 01-.214-.058c-.065-.043-.098-.108-.098-.196v-.359c0-.18-.013-.282-.04-.306-.058-.03-.158.033-.3.192a3.947 3.947 0 00-.405.56 3.91 3.91 0 01-.104.162c-.027.038-.048.057-.063.057-.012 0-.031-.01-.058-.028-.12-.035-.18-.103-.18-.203 0-.011.004-.036.012-.075a8.738 8.738 0 00.208-.821.059.059 0 00-.057-.034c-.039 0-.083.02-.133.058-.097.065-.22.221-.37.468-.189.289-.32.464-.393.525a.689.689 0 01-.075.11c-.02.023-.039.035-.058.035a.233.233 0 01-.086-.035c-.112-.065-.168-.158-.168-.277 0-.058.017-.174.052-.347.035-.173.073-.335.116-.485a2.29 2.29 0 00.138-.335c.016-.081.035-.168.058-.26a.488.488 0 01.052-.156.125.125 0 01.092-.041c.058 0 .1.031.127.093a.216.216 0 01.064.161.859.859 0 01-.064.283c-.065.143-.136.35-.213.624.08-.088.138-.157.173-.207.093-.131.212-.278.359-.44a.535.535 0 01.404-.196c.177.031.279.18.306.445 0 .047-.004.105-.012.174a1.92 1.92 0 01-.017.127 2.32 2.32 0 01.352-.388.398.398 0 01.237-.098c.062 0 .116.025.162.076.08.045.138.11.173.195a.788.788 0 01.053.312c0 .066-.002.118-.007.156a2.217 2.217 0 00-.011.145.64.64 0 00.04.22zm1.245.365a.537.537 0 01-.26-.191.63.63 0 01-.076-.336c0-.069.006-.15.017-.242l.012-.208a.21.21 0 00-.012-.08.165.165 0 00-.051-.064l-.058-.058c-.031-.031-.045-.056-.04-.075l-.007-.03a.22.22 0 01.012-.074c.011-.023.023-.034.035-.034a.208.208 0 00.046-.007.086.086 0 01.052-.005c.053 0 .08.011.08.034.016.031.041.035.076.012a.373.373 0 00.11-.15c.053-.085.132-.174.237-.266a1.59 1.59 0 01.265-.203.461.461 0 01.133-.04.688.688 0 01.162-.023c.07 0 .13.015.18.046a.742.742 0 01.161.133c.073.054.11.139.11.254 0 .081-.02.164-.058.249-.016.057-.078.13-.185.22-.104.088-.2.15-.289.185a1.55 1.55 0 01-.37.04.42.42 0 01-.179-.024c-.047-.027-.08.077-.098.312a.96.96 0 00-.005.11c0 .066.01.118.028.157l.041.046a.388.388 0 00.086.063.269.269 0 00.11.023.228.228 0 00.08-.011.346.346 0 00.123-.064c.038-.035.091-.087.16-.156.063-.061.107-.104.134-.127.02-.035.059-.052.116-.052.015 0 .042.004.08.012.035.027.052.055.052.086a.531.531 0 01-.057.173.71.71 0 01-.145.163 1.773 1.773 0 01-.196.156.745.745 0 01-.359.098.754.754 0 01-.253-.052zm.34-1.173c.039-.008.095-.018.168-.03a.356.356 0 00.18-.074c.05-.043.107-.114.173-.214a.264.264 0 00.017-.092c0-.058-.012-.099-.035-.122a.126.126 0 00-.092-.04.272.272 0 00-.168.07.533.533 0 00-.173.127 1.877 1.877 0 00-.191.219c-.053.069-.07.109-.052.121-.007.012.012.022.058.03.047.003.085.005.115.005zm3.26 1.103a.246.246 0 00-.057-.07c-.023-.027-.035-.044-.035-.051a.147.147 0 00-.029-.098c-.007-.008-.011-.025-.011-.052 0-.062.029-.166.086-.312.062-.15.141-.312.237-.485a.642.642 0 00.086-.122.561.561 0 01.076-.098c0-.015.006-.029.017-.04.012-.016.025-.024.04-.024v-.034c0-.035.064-.119.192-.25a1.72 1.72 0 01.312-.276.517.517 0 01.294-.098c.07 0 .13.019.18.057.019.023.107.107.265.25l.214.19-.104.288a.904.904 0 00-.052.283c0 .147.042.287.127.422a.31.31 0 00.115.122.435.435 0 00.173.07c.085.015.127.057.127.126 0 .077-.034.124-.103.14a.368.368 0 01-.07.005c-.227 0-.423-.16-.589-.48a.923.923 0 01-.104-.254 2.656 2.656 0 01-.185.219c-.42.42-.751.63-.993.63a.387.387 0 01-.209-.058zm.209-.323c0 .019.019.024.057.017a.516.516 0 00.15-.058c.062-.03.124-.07.186-.12.33-.285.57-.534.716-.746l.127-.209a.522.522 0 00-.121-.173c-.043-.038-.081-.057-.116-.057-.08 0-.208.102-.382.306-.173.2-.331.437-.473.71a1.96 1.96 0 00-.092.202zm4.843.179c-.043-.031-.097-.16-.162-.388a6.246 6.246 0 01-.144-.658 2.658 2.658 0 00-.093-.329.65.65 0 01-.023-.127c0-.07.029-.122.087-.156a.126.126 0 01.063-.023.171.171 0 01.075-.006c.085.03.162.265.232.704.011.112.028.212.051.301l.04.156.041.075c.016.023.034.035.053.035.011 0 .025-.008.04-.024.073-.073.152-.165.236-.277.093-.127.157-.211.191-.254a3.29 3.29 0 01.168-.23l.08-.11a1.27 1.27 0 01.122-.145c.039-.043.071-.068.098-.075a.177.177 0 01.053-.006c.065 0 .097.027.097.08 0 .05-.035.126-.104.226a.935.935 0 00-.074.127 2.585 2.585 0 01-.284.446 2.92 2.92 0 01-.226.288 1.178 1.178 0 01-.097.08.922.922 0 00-.15.168c-.124.116-.213.174-.266.174-.035 0-.07-.017-.104-.052zm2.332.37a.518.518 0 01-.41-.185.677.677 0 01-.162-.463c0-.185.04-.387.121-.606.085-.224.193-.424.324-.6.13-.177.263-.297.398-.36a.543.543 0 01.26-.062c.073 0 .157.02.25.062a.39.39 0 01.149.1.508.508 0 01.093.179c.011.069.017.116.017.143 0 .023-.01.073-.029.15a1.226 1.226 0 01-.185.365.959.959 0 01-.288.248.434.434 0 01-.232.058.668.668 0 01-.23-.04.376.376 0 01-.163-.133l-.04-.046-.018.046c-.08.208-.12.385-.12.53 0 .1.019.182.057.244.027.042.092.063.197.063a.443.443 0 00.167-.023.65.65 0 00.15-.069c.047-.031.091-.062.133-.093l.023-.011h-.006a1.373 1.373 0 00.179-.104.047.047 0 00.023-.006.07.07 0 00.024-.023c.034-.015.074-.01.12.017.047.027.07.062.07.105a.48.48 0 01-.104.15c-.061.07-.107.106-.138.11a.174.174 0 00-.04.017.252.252 0 00-.048.028h.007c-.035.047-.07.07-.104.07h-.047c-.015.03-.067.062-.156.093a.705.705 0 01-.242.046zm.26-1.272c.115 0 .221-.065.317-.196a.612.612 0 00.14-.37c0-.065-.014-.112-.041-.139a.148.148 0 00-.064-.017.493.493 0 00-.098.017c-.05.012-.127.066-.23.163a1.402 1.402 0 00-.232.248l-.075.121a.255.255 0 01.086-.017c.031-.004.049-.002.053.006a.21.21 0 00.04.132.125.125 0 00.104.052zm1.748 1.075a.31.31 0 01-.098-.023.281.281 0 01-.08-.04.182.182 0 01-.059-.133c0-.085.042-.187.127-.306.024-.047.082-.207.174-.48l.116-.318c.007-.065.003-.112-.012-.139-.015-.03-.048-.048-.098-.051h-.035c-.035 0-.058-.006-.07-.017-.011-.016-.017-.053-.017-.11a.188.188 0 01.035-.105.524.524 0 01.099-.08.413.413 0 01.19-.047c.062 0 .108.013.139.04.023.016.038.051.046.105.011.05.017.104.017.162.004.053.008.092.012.115.019-.02.065-.058.138-.115.077-.062.143-.12.197-.174.065-.047.14-.09.225-.132a.49.49 0 01.18-.064c.042 0 .084.011.127.034.042.02.07.045.086.076a.346.346 0 01.052.167.267.267 0 01-.063.162.359.359 0 01-.133.098.348.348 0 01-.116.035c-.023 0-.051-.017-.086-.052-.035-.035-.055-.064-.058-.087 0-.015-.05.015-.15.092a2.162 2.162 0 00-.226.192 1.765 1.765 0 00-.19.288l-.278.543-.08.196a.758.758 0 01-.053.133c-.015.023-.035.035-.058.035zm2.615.098c-.192 0-.332-.07-.42-.208a.166.166 0 01-.041-.115c0-.023.003-.043.01-.058.013-.02.026-.029.041-.029.02 0 .04.013.064.04.053.032.13.048.231.048.108 0 .214-.02.318-.059.123-.054.158-.096.104-.127h.005c-.15-.039-.3-.112-.45-.22-.147-.111-.227-.211-.242-.3a.58.58 0 01-.018-.145c0-.096.021-.192.064-.288a.986.986 0 01.173-.266c.115-.073.19-.114.225-.122v.007a.903.903 0 01.237-.168c.097-.047.22-.095.37-.144l.086-.012.116-.012a.072.072 0 00.029.012l.047.012a.134.134 0 00.057.017.285.285 0 01.19.132c.058.078.083.151.076.22a.442.442 0 01-.07.203c-.034.053-.076.084-.127.092.004-.004-.004-.012-.023-.023l-.046-.03-.017.012a.041.041 0 00-.006-.017.06.06 0 00-.012-.035.192.192 0 01-.017-.086l.006-.087.011-.058c0-.011-.008-.019-.023-.023a.138.138 0 00-.063-.012l-.047-.005a.699.699 0 00-.225.047.774.774 0 00-.22.114c-.13.085-.229.16-.294.226a.742.742 0 00-.15.219c-.047.066-.07.114-.07.145 0 .043.029.085.087.127.061.043.16.095.295.156.107.047.194.095.26.144.069.05.115.107.138.168a.121.121 0 01.006.047c0 .027-.004.044-.012.051.008.008.012.014.012.017v-.005a.185.185 0 01.023.075c0 .062-.038.12-.115.173a.811.811 0 01-.295.11h.006c-.093.027-.188.04-.284.04zm2.188-.057a.537.537 0 01-.26-.191.637.637 0 01-.075-.336c0-.069.006-.15.018-.242l.011-.208a.228.228 0 00-.011-.08.174.174 0 00-.053-.064l-.057-.058c-.031-.031-.045-.056-.04-.075l-.006-.03a.24.24 0 01.011-.074c.011-.023.023-.034.035-.034a.208.208 0 00.046-.007.084.084 0 01.052-.005c.054 0 .081.011.081.034.015.031.04.035.075.012a.366.366 0 00.109-.15c.055-.085.134-.174.238-.266.107-.097.196-.164.265-.203a.448.448 0 01.133-.04.688.688 0 01.162-.023c.07 0 .13.015.18.046a.723.723 0 01.16.133c.073.054.11.139.11.254a.605.605 0 01-.057.249c-.015.057-.077.13-.185.22-.104.088-.2.15-.288.185-.109.027-.232.04-.37.04a.424.424 0 01-.18-.024c-.046-.027-.079.077-.098.312a.796.796 0 00-.006.11c0 .066.01.118.029.157a.642.642 0 01.04.046.412.412 0 00.087.063.269.269 0 00.11.023.237.237 0 00.081-.011.34.34 0 00.12-.064c.04-.035.094-.087.163-.156.061-.061.106-.104.133-.127.02-.035.058-.052.115-.052.016 0 .043.004.081.012.035.027.053.055.053.086 0 .031-.02.089-.058.173a.715.715 0 01-.144.163 1.837 1.837 0 01-.197.156.745.745 0 01-.359.098.768.768 0 01-.254-.052zm.34-1.173l.168-.03a.356.356 0 00.18-.074c.05-.043.108-.114.173-.214a.264.264 0 00.017-.092c0-.058-.011-.099-.034-.122a.128.128 0 00-.093-.04.27.27 0 00-.167.07.541.541 0 00-.174.127 1.928 1.928 0 00-.19.219c-.054.069-.071.109-.052.121-.008.012.011.022.057.03.047.003.085.005.116.005z"
                  fill="none"
                />
              </clipPath>
            </defs>
            <path
              className="svg-logo__circlepath"
              d="M3.763.403C1.88.403.346 2.003.346 3.968c0 1.966 1.534 3.567 3.417 3.567 1.883 0 3.416-1.601 3.416-3.567 0-1.966-1.533-3.565-3.416-3.565zm0 .687c1.527 0 2.758 1.284 2.758 2.879 0 1.594-1.23 2.878-2.758 2.878-1.527 0-2.758-1.284-2.758-2.879 0-1.594 1.231-2.88 2.758-2.88z"
              fill="var(--color-brand)"
              fillOpacity="0.8"
              overflow="visible"
            />
            <path
              className="svg-logo__textpath"
              clipPath="url(#clip-path)"
              d="M6.02 2.606l.052.441-.134.45-.321 1.1V.746l.779.835.103.223-.241.142 1.087.51.04.989.03.634.123.163.178-.186 1.06-1.587 1.12.946.336.153.278-.42.164-.333-.1-.312-.375-.06-.338.301-.378.669-.142.606.13.305.388.158.825-.497 3.193-2.198-.089 1.121-.232.738-.091.45.136.069.126-.413.464-.45.439-.394.163.213-.05.377-.13.492.075.192.143-.452.585-.618.21.039.097.28.131 1.05.592-1.679.588.546.444.077.651-.174.1-.418-.195-.307-.404.074-.29.391-.114.27-.164.46.05.446.339.193.435-.168.369-.598 3.9-2.507.838 1.389-.592.518-.203-.375-.184-.084-.421.206-.367.483-.347.601-.064.333.243.146.462-.307.893-.895-.109.422.165.436.348.23.256.007 2.313-2.2.32.99.14.802.153.32.217.003.034-.325.663-.849.544-.578.54.806.358.297.278-.006.33-.34.126-.4s-.188-.222-.223-.254c-.036-.032-.4.088-.4.088l-.276.27-.316.54-.135.539.126.477.268.12.298-.068.666-.471.781-1.633.451.438.18-.03-.026.374-.076.499-.354.896.249-.877.375-.437.402-.384.325-.185.114.127-.192.093.081.086.252.166.68-1.375 2.29.219-.456 1.013-.248-.144-.067-.338-.377.092-.417.125-.275.186-.165.282-.084.276.228.304.295.147.255.147v.196l-.35.067-.375-.008-.437-.22 2.295-1.104.368.32.398.121.393.018.248-.157.178-.28-.029-.308-.23-.133-.294.035-.255.212-.276.37-.115.485.025.404.2.328.302.008.336-.19.346-.433"
              fill="none"
              stroke="var(--color-fg)"
              strokeWidth=".476"
            />
            <path
              d="M4.76 2.841a.291.291 0 01.036.133.278.278 0 01-.07.174l-.063.092a.84.84 0 00-.082.121.547.547 0 00-.051.145 9.633 9.633 0 01-.139.566l-.064.226-.092.358c-.173.597-.347.941-.52 1.034a.399.399 0 01-.144.029.5.5 0 01-.266-.087.737.737 0 01-.231-.226.974.974 0 01-.116-.26c-.035-.115-.04-.18-.018-.196.008-.007.022-.011.041-.011a.2.2 0 01.087.023c.03.019.05.04.058.063a.748.748 0 00.098.168c.05.07.087.106.11.11.061.03.113.046.155.046a.16.16 0 00.099-.035.608.608 0 00.156-.196c.047-.077.097-.203.15-.376l.047-.167c.092-.34.159-.56.201-.66.02-.06.04-.108.059-.143l-.105.092a2.764 2.764 0 01-.144.121.725.725 0 01-.156.093c-.027.034-.094.051-.203.051a.808.808 0 01-.23-.034.538.538 0 01-.15-.098.677.677 0 01-.082-.162.342.342 0 01-.023-.145c0-.165.039-.342.116-.53.073-.151.189-.311.347-.48.158-.17.295-.272.41-.307a.302.302 0 01.127-.033c.042 0 .086.008.132.023.062.007.11.02.146.04.034.02.092.063.173.132a.89.89 0 01.202.307zm-.635.573a.853.853 0 00.168-.59.268.268 0 00-.024-.11c-.011-.023-.032-.034-.063-.034a.461.461 0 00-.254.103c-.096.07-.189.161-.278.273a1.055 1.055 0 00-.144.276c-.047.131-.07.253-.07.365 0 .085.016.133.047.144a.594.594 0 00.323-.104c.1-.065.198-.173.295-.323z"
              fill="var(--color-fg)"
            />
          </svg>
        </a>
      </div>
      <div
        className="header__menu-icon-wrapper"
      >
        <div
          aria-label="Open navigation menu"
          className="clickable"
          id="header__menu-icon"
          onClick={[Function]}
          onKeyUp={[Function]}
          role="button"
          tabIndex={0}
        >
          <svg
            aria-hidden="true"
            className="svg-menu"
            height="24"
            viewBox="0 0 9 18"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M-7.5 0h24v24h-24z"
              fill="none"
            />
            <path
              className="svg-menu__path"
              d="M-4.5 15h18v-2h-18zm0-5h18V8h-18zm0-7v2h18V3z"
              fill="var(--color-mid-fg)"
            />
          </svg>
        </div>
      </div>
    </div>
    <nav
      id="nav"
    >
      <div
        className="nav__header"
      >
        <div
          className="nav__close-icon-wrapper"
        >
          <div
            aria-hidden="true"
            aria-label="Close navigation menu"
            className="clickable  nav__close-icon"
            onClick={[Function]}
            onKeyUp={[Function]}
            role="button"
            tabIndex={-1}
          >
            <svg
              aria-hidden="true"
              className="svg-close"
              height="24"
              viewBox="0 0 14 14"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M-5-5h24v24H-5z"
                fill="none"
              />
              <path
                className="svg-close__path"
                d="M14 1.41L12.59 0 7 5.59 1.41 0 0 1.41 5.59 7 0 12.59 1.41 14 7 8.41 12.59 14 14 12.59 8.41 7z"
                fill="var(--color-mid)"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        className="nav__block"
      >
        <ul
          className="nav__list"
        >
          <li
            className="nav__item"
          >
            <a
              aria-hidden="true"
              aria-label="Click here to navigate to the Home page."
              className="clickable  nav__link"
              href="/"
              id="home-link"
              onClick={[Function]}
              tabIndex={-1}
            >
              Home
            </a>
          </li>
          <li
            className="nav__item"
          >
            <a
              aria-hidden="true"
              aria-label="Click here to navigate to the About page."
              className="clickable  nav__link"
              href="/about"
              id="about-link"
              onClick={[Function]}
              tabIndex={-1}
            >
              About
            </a>
          </li>
          <li
            className="nav__item"
          >
            <a
              aria-hidden="true"
              aria-label="Click here to navigate to the Contact page."
              className="clickable  nav__link"
              href="/contact"
              id="contact-link"
              onClick={[Function]}
              tabIndex={-1}
            >
              Contact
            </a>
          </li>
        </ul>
      </div>
    </nav>
  </header>,
  <main
    className="page-not-found"
    id="main"
  >
    <section
      className="page-not-found__content"
    >
      <h1
        className="page-not-found__title"
      >
        404 — Page Not Found
      </h1>
      <p
        className="page-not-found__text"
      >
        Sorry, this page doesn’t exist! Go back to the
         
        <a
          className="clickable  page-not-found__link"
          href="/"
          onClick={[Function]}
        >
          home page
        </a>
        .
      </p>
    </section>
  </main>,
]
`;
